### PR TITLE
Reorder locals.

### PR DESCRIPTION
**Pull request by mbebenita**
Sorts locals by access frequency.

### CODE REVIEW

File: src/passes/ReorderLocals.cpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+#include <wasm.h>
+#include <pass.h>
+
+namespace wasm {
+
+struct ReorderLocals : public WalkerPass<WasmWalker<ReorderLocals, void>> {
+
+  std::map<Name, uint32_t> counts;
+
+  void finalize(PassRunner* runner, Module *module) override {
```
Comment (kripken) [inline src/passes/ReorderLocals.cpp]: why do we need `finalize`?

**Comment by kripken** [comment]
lgtm aside from that one question, and something seems wrong on travis.

**Comment by kripken** [comment]
The travis problem looks weird, didn't you update that test? I'll try locally.

**Comment by mbebenita** [comment]
I don't know what is going on in Travis. Tests run okay locally, and I updated the `.fromasm` files using `./auto_update_tests.py`.

File: src/passes/ReorderLocals.cpp (hunk 1 above)
Comment (mbebenita) [inline src/passes/ReorderLocals.cpp]: We don't :)

**Comment by kripken** [comment]
I think it's a stable sort issue, I see the errors locally. I might have a fix, will merge with it if so.

**Comment by kripken** [comment]
Pushed with fix.

Thanks!

**Pull request merged**